File: src/main/java/org/example/kalkulationsprogramm/domain/AnfrageNotizBild.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * Bild zu einer Anfrage-Notiz.
 * Ermöglicht 1:n Beziehung zwischen Notiz und Bildern.
 */
@Getter
@Setter
@Entity
@Table(name = "anfrage_notiz_bild")
public class AnfrageNotizBild {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "notiz_id", nullable = false)
    private AnfrageNotiz notiz;

    @Column(nullable = false)
    private String gespeicherterDateiname;

    @Column
    private String originalDateiname;

    @Column
    private String dateityp;

    @Column(nullable = false)
    private LocalDateTime erstelltAm;

    @PrePersist
    protected void onCreate() {
        if (erstelltAm == null) {
            erstelltAm = LocalDateTime.now();
        }
    }
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/AnfrageNotizBild.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "anfrage_notiz_bild")
open class AnfrageNotizBild {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "notiz_id", nullable = false)
    open var notiz: AnfrageNotiz? = null

    @Column(nullable = false)
    open var gespeicherterDateiname: String? = null

    @Column
    open var originalDateiname: String? = null

    @Column
    open var dateityp: String? = null

    @Column(nullable = false)
    open var erstelltAm: LocalDateTime? = null

    @PrePersist
    open fun onCreate() {
        if (erstelltAm == null) erstelltAm = LocalDateTime.now()
    }

}